No such file or directory after upgrading to Gradle 4.2

I think this exception is related to `gradle-git-publish` when running on Gradle 4.2 -- or am I mistaken?

```
Caused by: org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of add command
	at org.eclipse.jgit.api.AddCommand.call(AddCommand.java:253)
	at org.eclipse.jgit.api.AddCommand.call(AddCommand.java:85)
	at java_util_concurrent_Callable$call$0.call(Unknown Source)
	at org.ajoberstar.grgit.operation.AddOp.call(AddOp.groovy:53)
	at org.ajoberstar.grgit.operation.AddOp.call(AddOp.groovy)
	at java_util_concurrent_Callable$call.call(Unknown Source)
	at org.ajoberstar.grgit.internal.OpSyntax.mapOperation(OpSyntax.groovy:33)
	at org.ajoberstar.grgit.internal.OpSyntax$mapOperation.callStatic(Unknown Source)
	at org.ajoberstar.grgit.Grgit.add(Grgit.groovy)
	at org.ajoberstar.grgit.Grgit$add$7.call(Unknown Source)
	at org.ajoberstar.gradle.git.publish.GitPublishPlugin$_createCommitTask_closure5$_closure15.doCall(GitPublishPlugin.groovy:141)
	at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:700)
	at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:673)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:121)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
	... 94 more
Caused by: java.io.IOException: No such file or directory
	at org.eclipse.jgit.internal.storage.file.ObjectDirectoryInserter.newTempFile(ObjectDirectoryInserter.java:273)
	at org.eclipse.jgit.internal.storage.file.ObjectDirectoryInserter.toTemp(ObjectDirectoryInserter.java:200)
	at org.eclipse.jgit.internal.storage.file.ObjectDirectoryInserter.insert(ObjectDirectoryInserter.java:144)

```

See log after the upgrade to Gradle 4.2 with complete stacktrace at https://junit.ci.cloudbees.com/blue/organizations/jenkins/JUnit5/detail/JUnit5/2230/pipeline

Or this is the latest one with more debug information:
https://junit.ci.cloudbees.com/blue/rest/organizations/jenkins/pipelines/JUnit5/runs/2260/log/?start=0

--- a/src/main/groovy/org/ajoberstar/gradle/git/publish/GitPublishPlugin.groovy
+++ b/src/main/groovy/org/ajoberstar/gradle/git/publish/GitPublishPlugin.groovy
@@ -71,6 +71,7 @@ class GitPublishPlugin implements Plugin<Project> {
     task.with {
       group = 'publishing'
       description = 'Prepares a git repo for new content to be generated.'
+
       // get the repo in place
       doFirst {
         Grgit repo = findExistingRepo(project, extension).orElseGet { freshRepo(extension) }
@@ -121,12 +122,16 @@ class GitPublishPlugin implements Plugin<Project> {
   }
 
   private Task createCopyTask(Project project, GitPublishExtension extension) {
-    Task task = project.tasks.create(COPY_TASK, Copy)
+    Task task = project.tasks.create(COPY_TASK)
     task.with {
       group = 'publishing'
       description = 'Copy contents to be published to git.'
-      with extension.contents
-      into { extension.repoDir }
+      doLast {
+        project.copy {
+          with extension.contents
+          into { extension.repoDir }
+        }
+      }
     }
     return task
   }

